libs/blocks/merch-card/merch-card.js

```diff
@@ -350,11 +369,20 @@ const init = async (el) => {
     footer.append(ctas);
   }
   merchCard.appendChild(footer);
-  const offerSelection = cardType === 'plans' ? el.querySelector('ul') : null;
-  if (offerSelection) {
-    const { initOfferSelection } = await import('./merch-offer-select.js');
-    initOfferSelection(merchCard, offerSelection);
+
+  if (cardType === 'plans') {
+    const quantitySelect = createQuantitySelect(el);
+    const offerSelection = el.querySelector('ul');
```

Comment:
Here you are again looking for any `ul` and if a card has text content with this tag it will cause unexpected behavior. 
I suggest to refactor `createQuantitySelect(el)` to initialize respectively without introducing variant-specific code inside `init` and avoid running the same `querySelector()` more than once. I feel like everything quantity select-related could be inside one function.

Reply:
This is a different component, so that is why  I need to query it again.